viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -19,6 +19,44 @@
 #include "cartographer/mapping/map_builder.h"
 #include "glog/logging.h"
 
+namespace {
+// The default value when coloring the image if no data is present. A
+// pixel representing no data will have all 3 channels of its color channels
+// (RGB) as the default value([102,102,102])
+static const unsigned char defaultNoDataRGBValue = 102;
+// Number of bytes in a pixel
+static const int bytesPerPixel = 4;
+struct ColorARGB {
+    unsigned char A;
+    unsigned char R;
+    unsigned char G;
+    unsigned char B;
+};
+
+// Check if pixel is the default color signaling no data is present
+bool CheckIfEmptyPixel(ColorARGB pixel_color) {
+    return (pixel_color.R == defaultNoDataRGBValue) &&
+           (pixel_color.G == defaultNoDataRGBValue) &&
+           (pixel_color.B == defaultNoDataRGBValue);
+}
+
+// Conver the scale of a specified pixel color channel from the given
+// 102-255 range to 100-0. This is an initial solution for extracting
+// probability information from cartographer
+int CalculateProbabilityFromColorChannels(ColorARGB pixel_color) {
+    unsigned char max_val = CHAR_MAX;
```

Comment:
This is equal to 127.

I think you might want `UCHAR_MAX`?

Also, if it is important that it be from 102-255... maybe this can just be another constant like `maxDataRGBValue` which we set to 255?

Reply:
![Uploading Screenshot 2023-04-03 at 4.00.11 PM.png…]()

Reply:
This is correct, I checked the max value in a channel for a large map run and recorded 253. Changing this to UCHAR_MAX